src/code/components/clutch-view.js

```diff
@@ -40,48 +29,49 @@ const ClutchView = React.createClass({
     if (this.props.onClick) this.props.onClick(index, id, org);
   },
 
-  handlePageForward() {
-    this.setState({pagesBack: Math.max(this.state.pagesBack - 1, 0)});
+  handlePageBackward() {
+    this.setState({page: Math.max(this.state.page - 1, 1)});
   },
 
-  handlePageBackward() {
+  handlePageForward() {
     const numPages = this.props.orgs.length / 8;
-    this.setState({pagesBack: Math.min(this.state.pagesBack + 1, numPages - 1)});
+    this.setState({page: Math.min(this.state.page + 1, numPages)});
   },
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.orgs.length === 0) {
-      this.setState({pagesBack: 0});
+      this.setState({page: 0});
     }
     if (nextProps.orgs.length > this.props.orgs.length) {
       // Move to the end whenever a new clutch is bred
-      this.setState({pagesBack: 0});
+      const numPages = nextProps .orgs.length / 8;
+      this.setState({page: numPages});
     }
   },
 
   render() {
-    let displayStyle = {size: this.props.height*(5/8), top: -153, marginLeft: -12},
-        stableDrakeViews = this.props.orgs.map((org, index) => {
+    let displayStyle = {size: 136, top: -153, marginLeft: -12},
+        drakeViews = this.props.orgs.map((org, index) => {
```

Comment:
Instead of creating all the views, and then slicing them, why not slice the drake list first and then create the views?

```
let firstDrake = (this.props.orgs.length - 8) - (this.state.pagesBack * 8),
    visibleDrakes = this.props.orgs.slice(firstDrake, firstDrake + 8),
    drakeViews = visibleDrakes.map(....
```

it probably doesn't make a huge amount of difference in terms of speed, because if you don't add the views then no DOM elements will be created (though you do create the views, so if you have hundreds of offspring it might become noticeable), but it just seems to me to be more easily understandable.